packages/common/core-interfaces/src/test/jsonStringifyAndParse.spec.ts

```diff
@@ -0,0 +1,2701 @@
+				const { stringified: resultNumeric, out: outNumeric } =
+					stringifyThenParse(NumericEnum);
+				assertIdenticalTypes(
+					resultNumeric,
+					createInstanceOf<JsonString<typeof NumericEnum>>(),
+				);
+				assertIdenticalTypes(outNumeric, NumericEnum);
+				const { stringified: resultString, out: outString } = stringifyThenParse(StringEnum);
+				assertIdenticalTypes(resultString, createInstanceOf<JsonString<typeof StringEnum>>());
+				assertIdenticalTypes(outString, StringEnum);
+				const { stringified: resultComputed, out: outComputed } =
+					stringifyThenParse(ComputedEnum);
+				assertIdenticalTypes(
+					resultComputed,
+					createInstanceOf<JsonString<typeof ComputedEnum>>(),
+				);
+				assertIdenticalTypes(outComputed, ComputedEnum);
+			});
+
+			it("object with `readonly`", () => {
+				const { stringified, out } = stringifyThenParse(objectWithReadonly);
+				assertIdenticalTypes(
+					stringified,
+					createInstanceOf<JsonString<typeof objectWithReadonly>>(),
+				);
+				assertIdenticalTypes(out, objectWithReadonly);
+			});
+
+			it("object with getter implemented via value", () => {
+				const { stringified, out } = stringifyThenParse(objectWithGetterViaValue);
+				assertIdenticalTypes(
+					stringified,
+					createInstanceOf<JsonString<typeof objectWithGetterViaValue>>(),
+				);
+				assertIdenticalTypes(out, objectWithGetterViaValue);
+			});
+			it("object with setter implemented via value", () => {
+				const { stringified, out } = stringifyThenParse(objectWithSetterViaValue);
+				assertIdenticalTypes(
+					stringified,
+					createInstanceOf<JsonString<typeof objectWithSetterViaValue>>(),
+				);
+				assertIdenticalTypes(out, objectWithSetterViaValue);
+			});
+			it("object with matched getter and setter implemented via value", () => {
+				const { stringified, out } = stringifyThenParse(
+					objectWithMatchedGetterAndSetterPropertyViaValue,
+				);
+				assertIdenticalTypes(
+					stringified,
+					createInstanceOf<
+						JsonString<typeof objectWithMatchedGetterAndSetterPropertyViaValue>
+					>(),
+				);
+				assertIdenticalTypes(out, objectWithMatchedGetterAndSetterPropertyViaValue);
+			});
+			it("object with mismatched getter and setter implemented via value", () => {
+				const { stringified, out } = stringifyThenParse(
+					objectWithMismatchedGetterAndSetterPropertyViaValue,
+				);
+				assertIdenticalTypes(
+					stringified,
+					createInstanceOf<
+						JsonString<typeof objectWithMismatchedGetterAndSetterPropertyViaValue>
+					>(),
+				);
+				assertIdenticalTypes(out, objectWithMismatchedGetterAndSetterPropertyViaValue);
+			});
+
+			describe("class instance", () => {
+				it("with public data (just cares about data)", () => {
+					const { stringified, out } = stringifyThenParse(classInstanceWithPublicData, {
+						public: "public",
+					});
+					assertIdenticalTypes(
+						stringified,
+						createInstanceOf<JsonString<typeof classInstanceWithPublicData>>(),
```

Comment:
Or this?

```
JsonString<ClassWithPublicData>
```

Also, I'm not sure how to think about `JsonString<SomeClass>` since the conversion is lossy. Meaning, `out: ClassWithPublicData` but... is it really?  What would `out instanceof ClassWithPublicData` say?

Reply:
Or just move this case to the "known defects" section